lib/beaker/hypervisor.rb

```diff
@@ -57,6 +58,26 @@ def cleanup
       nil
     end
 
+    DEFAULT_CONNECTION_PREFERENCE = [:ip, :vmhostname, :hostname]
+    # SSH connection method preference. Can be overwritten by hypervisor to change the order
+    def connection_preference(host)
+      DEFAULT_CONNECTION_PREFERENCE
+    end
+
+    def self.set_ssh_connection_preference(hosts_to_provision, hypervisor)
+      hosts_to_provision.each do |host|
+        ssh_methods = hypervisor.connection_preference(host)
```

Comment:
is it cleaner to change this line to `ssh_methods = hypervisor.connection_preference(host) + DEFAULT_CONNECTION_PREFERENCE`